File: src/app/conjugation/forms/non-past-short-affirmative.ts
import {Conjugation, Word, WordType} from "../conjugation";

export class NonPastShortAffirmative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {
        switch (word.wordType) {

            // Verbs
            case WordType.GodanVerb:
                return word;

            case WordType.IchidanVerb:
                return word;

            case WordType.SuruVerb:
                return word;

            case WordType.KuruVerb:
                return word;


            // Adjectives
            case WordType.IAdjective:
                if (word.kanji === '良い') {
                    return word.replace(word.kanji, 'いい');
                }
                return word;

            case WordType.NaAdjective:
                return word;


            default:
                return undefined;
        }
    }

    getTitle(): string {
        return "Non-past, short, affirmative"
    }

    getSettingsTitle(): string {
        return "Short, affirmative"
    }

}

File: backend/app/conjugation/forms/non_past_short_affirmative.py
from ..core import Conjugation, Word, WordType


class NonPastShortAffirmative(Conjugation):

    title = 'Non-past, short, affirmative'
    settings_title = 'Short, affirmative'

    def conjugate(self, word: Word) -> Word | None:
        match word.word_type:

            # Verbs
            case (
                WordType.GODAN_VERB
                | WordType.ICHIDAN_VERB
                | WordType.SURU_VERB
                | WordType.KURU_VERB
            ):
                return word

            # Adjectives
            case WordType.I_ADJECTIVE:
                if word.kanji == '良い':
                    return word.replace(word.kanji, 'いい')

                return word

            case WordType.NA_ADJECTIVE:
                return word

            case _:
                return None
